tiny_dnn/layers/deconvolutional_layer.h

```diff
@@ -198,7 +214,12 @@ class deconvolutional_layer : public layer {
                         size_t w_stride        = 1,
                         size_t h_stride        = 1,
                         backend_t backend_type = core::default_engine())
-    : layer(std_input_order(has_bias), {vector_type::data}) {
+    : layer({vector_type::data}, {vector_type::data}) {
+    layer::add_parameter(out_channels, in_channels, window_height, window_width,
+                         parameter_type::weight);
```

Comment:
Yes it essentially allocates memory to data and grad depending on the dimensions. But it allocates memory to `Parameter` struct, I think `xtensor` itself stays uninitialized unless called upon.

Unfortunately I don't clearly remember how it worked out, @Randl a little help here ? 
Here is `add_parameter` body to save you from navigating:
```c++
  /** @brief Add new parameter to this layer, to be called in constructor. */
  void add_parameter(size_t out_channels,
                     size_t in_channels,
                     size_t height,
                     size_t width,
                     parameter_type type,
                     bool trainable = true) {
    parameters_.push_back(std::make_shared<Parameter>(
      out_channels, in_channels, height, width, type, trainable));
  }
```